photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/NameAndTerm.scala

```diff
@@ -14,37 +14,51 @@
  */
 package com.linkedin.photon.ml.data.avro
 
+import com.linkedin.photon.ml.util.Summarizable
+
 /**
  * A compact way to represent the feature key as (name, term) pair.
+ *
+ * TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]]
+ *       after Avro related classes/functions are decoupled from the rest of code
  */
-// TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]]
-// after Avro related classes/functions are decoupled from the rest of code
-protected[ml] case class NameAndTerm(name: String, term: String) {
+protected[ml] case class NameAndTerm(name: String, term: String) extends Summarizable {
+
+  /**
+   * Build a human-readable summary for the [[NameAndTerm]].
+   *
+   * @return A summary of the [[NameAndTerm]] in string representation
+   */
+  override def toSummaryString: String = s"name: $name, term: $term"
 
   /**
+   * Returns a string representation of the [[NameAndTerm]]
    *
-   * @param that
-   * @return
+   * @return A string representation of the name-term pair
+   */
+  override def toString: String = s"$name\t$term"
```

Comment:
Only if it they contain tabs - this is only used when creating the feature bags so it doesn't actually need to be human readable. If you've got suggestions, I can implement them now - otherwise I'd prefer to keep out of the Avro mess